### DISCUSSION TITLE
How can we pass the test-flags while running "xcode-project run-tests" command?

### DISCUSSION THREAD

**Discussion by SamarthKejriwal** [codemagic-ci-cd/codemagic-docs#2003] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

I read the [documentation](https://docs.codemagic.io/yaml-testing/testing/)  and found the use case of `test-xcargs` when running the `xcode-project run-tests` command. 
Checking the whole [CLI doc](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/xcode-project/run-tests.md#run-tests) also I was not able to figure out the syntax to write the `test-flags` in the command.

Command That is tried was - 

`xcode-project run-tests --workspace "ABC.xcworkspace" --scheme "ABCUITests" --device "iOS 16.2 iPhone 14" --test-only "ABCUITests" --test-flags "-skip-testing=ABCUITests/ABCLaunchTimeTest" --verbose.`

### Expected results

The command should run successfully

### Actual results

The following error was shown - 

```
usage: xcode-project run-tests [-h] [--log-stream {stderr,stdout}] [--no-color] [--version] [-s] [-v] [--project XCODE_PROJECT_PATH]
                               [--workspace XCODE_WORKSPACE_PATH] [--target TARGET_NAME] [--config CONFIGURATION_NAME]
                               [--scheme SCHEME_NAME] [--clean] [--disable-coverage] [--graceful-exit] [--max-concurrent-devices MAX_CONCURRENT_DEVICES]
                               [--max-concurrent-simulators MAX_CONCURRENT_SIMULATORS] [-d device_description [device_description ...]]
                               [--test-only TEST_ONLY] [--sdk TEST_SDK] [--omit-sdk] [-o OUTPUT_DIR] [-e OUTPUT_EXTENSION]
                               [--test-flags TEST_FLAGS] [--test-xcargs TEST_XCARGS] [--disable-xcpretty] [--xcpretty-options XCPRETTY_OPTIONS]
xcode-project run-tests: error: argument --test-flags: expected one argument

```

**Comment by himesh-cm** [comment]
Hi @SamarthKejriwal, Let me check with the devs and I will keep you posted.

**Reply by himesh-cm** [reply]
Hi @SamarthKejriwal 
Please remove `-` and add it like --test-flags `skip-testing=ABCUITests/ABCLaunchTimeTest`.
Let me know, if you are still facing any issues.

**Reply by SamarthKejriwal** [reply]
Sure will check and confirm

**Comment by himesh-cm** [comment]
Hi @SamarthKejriwal, I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new one.

**Reply by SamarthKejriwal** [reply]
@himesh-cm I tried the following command, it gave me error - 

```
xcode-project run-tests \
          --workspace "$XCODE_WORKSPACE" \
          --scheme "$UITEST_SCHEME" \
          --device "$DESTINATION_UITEST_1" \
          --test-only "$UITEST_SCHEME" \ 
          --test-flags "resultBundlePath=build/TestResults skip-testing=TestA skip-testing=TestAB skip-testing=TestC skip-testing=TestD" \
          --verbose
```
PS: I am trying to pass multiple test flags, is it the correct way to do it?

Error was - 

```
xcode-project: error: unrecognized arguments:  
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_11_1_13xkm16d: line 12: --test-flags: command not found
/var/folders/w2/rrf5p87d1bbfyphxc7jdnyvh0000gn/T/build_script_11_1_13xkm16d: line 13: --verbose: command not found
```

**Reply by SamarthKejriwal** [reply]
@himesh-cm any updates on this?
